File: WebApi/Classes/TurnResult.vb
Imports Camera
Imports Entities
Imports Scene

Namespace Classes
    Public Class TurnResult

        Private ReadOnly _sceneService As ISceneService
        Private ReadOnly _cameraService As ICameraService
        Private ReadOnly _camera As Camera

        Public Sub New(sceneService As ISceneService,
                       cameraService As ICameraService)
            _sceneService = sceneService
            _cameraService = cameraService
            _camera = New Camera(_cameraService)
        End Sub

        Public ReadOnly Property Scene As List(Of BaseGameEntity)
            Get
                Return _sceneService.CurrentSceneState
            End Get
        End Property

        Public ReadOnly Property Camera As Camera
            Get
                Return _camera
            End Get
        End Property

    End Class
End Namespace

File: space-game/WebApi/Classes/TurnResult.cs
using Camera;
using Scene;

namespace WebApi.Classes
{
    public class TurnResult
    {
        private readonly ISceneService _sceneService;
        private readonly ICameraService _cameraService;

        public TurnResult(ISceneService sceneService, ICameraService cameraService)
        {
            _sceneService = sceneService;
            _cameraService = cameraService;
            Camera = new Camera(_cameraService);
        }

        public ISceneState Scene => _sceneService.CurrentSceneState;

        public Camera Camera { get; }
    }
}
